src/build/tomcat80-compat.groovy

```diff
@@ -0,0 +1,46 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import org.objectweb.asm.ClassReader
+import org.objectweb.asm.ClassVisitor
+import org.objectweb.asm.ClassWriter
+import org.objectweb.asm.MethodVisitor
+import org.objectweb.asm.Opcodes
+
+import java.nio.file.Files
+import java.nio.file.Paths
+
+// This script removes the methods that did not exist until Tomcat 8.5 from the ProtocolHandler implementation.
```

Comment:
```java
// We need the methods to exist during Java compilation since we compile both classes against tomcat 8.5, but want to make sure the bytecode doesn't exist during classload in the legacy version where they don't exist.
```

I think a better approach might be providing a support library with no-op implementations of the missing classes, that users of legacy tomcat can just add a dependency on. This would be easier after #185 though